components/post_view/post_add_bot_teams_channels/post_add_bot_teams_channels.jsx

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {Client4} from 'mattermost-redux/client';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import GenericTeamChannelProvider from 'components/suggestion/generic_team_channel_provider.jsx';
+import MenuActionProvider from 'components/suggestion/menu_action_provider';
+import AutocompleteSelector from 'components/widgets/settings/autocomplete_selector';
+import PostEmoji from 'components/post_emoji';
+
+import {localizeMessage} from 'utils/utils.jsx';
+
+export default class PostAddBotTeamsChannels extends React.PureComponent {
+    static propTypes = {
+
+        /*
+        * Post containing the user id of the bot
+        */
+        post: PropTypes.object.isRequired,
+
+        /*
+        * List of teams available to the current user
+        */
+        teams: PropTypes.arrayOf(PropTypes.object).isRequired,
+
+        actions: PropTypes.shape({
+
+            /*
+            * Function to add members to channel
+            */
+            addChannelMember: PropTypes.func.isRequired,
+
+            /*
+            * Function to add user to team
+            */
+            addUserToTeam: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {addMode: false};
+    }
+
+    handleAddButton = () => {
+        this.setState({addMode: true});
+    }
+
+    handleSelectedTeam = (selected) => {
+        this.setState({selectedTeamName: selected.text, team: selected});
+    }
+
+    handleSelectedChannel = (selected) => {
+        this.setState({selectedChannelName: selected.display_name, channel: selected});
+
+        return this.props.actions.addUserToTeam(this.state.team.value, this.props.post.user_id).then(
+            () => this.props.actions.addChannelMember(this.state.channel.id, this.props.post.user_id)).then(
```

Comment:
If you used async/await here you could make this function a lot cleaner and easier to read

Reply:
Wow that is a lot cleaner! Fixed!